packages/search-ui/src/A11yNotifications.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * This helper creates a live region that announces the results of certain
+ * actions (e.g. searching, paging, etc.), that are otherwise invisible
+ * to screen reader users.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions
+ */
+const regionId = "search-ui-screen-reader-notifications";
+const hasDOM = typeof document !== "undefined"; // Prevent errors in SSR apps
+
+const getLiveRegion = () => {
+  if (!hasDOM) return;
+
+  let region = document.getElementById(regionId);
+  if (region) return region;
+
+  region = document.createElement("div");
+  region.id = regionId;
+  region.setAttribute("role", "status");
+  region.setAttribute("aria-live", "polite");
+
+  /**
+   * Visually-hidden CSS that's still available to screen readers.
+   * We're avoiding putting this in a stylesheet to ensure that this
+   * still works for users that opt for custom views & CSS. We're
+   * also opting to use CSSOM instead of inline styles to avoid
+   * Content Security Policy warnings.
+   *
+   * @see https://accessibility.18f.gov/hidden-content/
+   */
+  region.style.position = "absolute";
+  region.style.width = "1px";
+  region.style.height = "1px";
+  region.style.margin = "-1px";
+  region.style.padding = "0";
+  region.style.border = "0";
+  region.style.overflow = "hidden";
+  region.style.clip = "rect(0 0 0 0)";
+
+  document.body.appendChild(region);
+  return region;
+};
+
+const announceToScreenReader = announcement => {
+  if (hasDOM) {
+    const region = getLiveRegion();
+    region.textContent = announcement;
+  }
+};
+
+const defaultMessages = {
+  searchResults: ({ start, end, totalResults, searchTerm }) => {
+    let message = `Showing ${start} to ${end} results out of ${totalResults}`;
+    if (searchTerm) message += `, searching for "${searchTerm}".`;
+    return message;
+  },
+  moreFilters: ({ optionsCount, showingAll }) => {
```

Comment:
Yeah, I see your point.

Reply:
Hm, I see what you're saying about it feeling weird that `search-ui` contains a default phrase that's only relevant to `react-search-ui`. I'm not _incredibly_ bothered by it but it does make sense that someone only using Search UI vanilla w/o React would find that message function to be useless/cruft.

Maybe what we could do is move the `moreFilters` message function with `react-search-ui`'s SearchDriver() setup here:

https://github.com/elastic/search-ui/blob/da835da856df9c86641cd463a6c462803dcbf472/packages/react-search-ui/src/SearchProvider.js#L26-L35

Could instead become:

```jsx
  componentDidMount() {
    // This could possibly go into its own `react-search-ui/A11yNotifications.js` file
    const defaultMessages = {
      moreFilters: ({ visibleOptionsCount, showingAll }) => {
        let message = showingAll ? "All " : "";
        message += `${visibleOptionsCount} options shown.`;
        return message;
      }
    };

    const { config } = this.props;
    config.a11yNotificationMessages = {
      ...defaultMessages,
      ...config.a11yNotificationMessages
    };

    // This initialization is done inside of componentDidMount, because initializing the SearchDriver server side
    // will error out, since the driver depends on window. Placing the initialization inside of componentDidMount
    // assures that it won't attempt to initialize server side.
    const driver = new SearchDriver(config);
    this.setState({
      driver
    });
  }
```

Thoughts?

Reply:
That's a good thought. But I guess Vadim's concern still applies if you think about the library in 3 layers:

- low-level vanilla JS API
- low-level React API <- more filters message
- React components <- more filters concern

Reply:
That's a really good idea, Constance. But Jason has a good point: it would still split "more filters" logic between two packages, one of which could be used without the other.

Reply:
Maybe I'm not understanding the concern fully? To me it makes sense that different parts of the layers need access to accessible notifications, the same way that react and react views all still deal with SearchDriver in one way or another.

I'm also not 100% seeing:
> it would still split "more filters" logic between two packages, one of which could be used without the other
Which two packages are these? We're not touching anything in `react-search-ui-views` right now, only `react-search-ui`, so there actually isn't a split there (that wasn't already there before). Developers can still roll their own custom Filter/Facet view with a custom `+ More` button and still have an a11y notification for `onMoreClick`, which I think is ideal.

Maybe we can hop on a call sometime today to clarify this (since it's a pretty architecturally large concern)?

Reply:
You're right. I thought that would split the logic between two packages, but I was wrong. I'm not concerned about the fact that the logic is split between low-level React API and components.
In conclusion, let's do this!

Reply:
This has been addressed in 70d8379!